assimilation_code/modules/io/io_filenames_mod.f90

```diff
@@ -632,24 +640,39 @@ subroutine check_attributes(ncFile, filename, ncVarId, domid, varid)
 if ( get_has_missing_value(domid, varid) ) then
    select case (get_xtype(domid,varid))
       case (NF90_INT)
-         call get_FillValue(domid, varid, spvalINT)
-         call check_attribute_value_int(ncFile, filename, ncVarID, '_FillValue', spvalINT)
          call get_missing_value(domid, varid, spvalINT)
          call check_attribute_value_int(ncFile, filename, ncVarID, 'missing_value', spvalINT)
 
       case (NF90_FLOAT)
-         call get_FillValue(domid, varid, spvalR4)
-         call check_attribute_value_r4(ncFile, filename, ncVarID, '_FillValue', spvalR4)
          call get_missing_value(domid, varid, spvalR4)
          call check_attribute_value_r4(ncFile, filename, ncVarID, 'missing_value', spvalR4)
 
       case (NF90_DOUBLE)
-         call get_FillValue(domid, varid, spvalR8)
-         call check_attribute_value_r8(ncFile, filename, ncVarID, '_FillValue', spvalR8)
          call get_missing_value(domid, varid, spvalR8)
          call check_attribute_value_r8(ncFile, filename, ncVarID, 'missing_value', spvalR8)
 
       case default
+         !>@todo FIXME report the variable with the unsupported xtype
+         call error_handler(E_ERR, 'check_attributes', 'unknown xtype', source)
+   end select
+endif
+
+if ( get_has_FillValue(   domid, varid) ) then
```

Comment:
are these routines used in model_to_dart and dart_to_model?   because i thought the idea was to get away from having to support either missing values or fill values.  this looks like it's adding more code to support them.